picnic/src/main/java/com/jakewharton/picnic/textLayout.kt

```diff
@@ -53,10 +59,12 @@ internal class SimpleLayout(private val cell: PositionedCell) : TextLayout {
 
     var x = left
     var y = top
-    for (char in cell.cell.content) {
-      // TODO invisible chars, codepoints, graphemes, etc.
-      if (char != '\n') {
-        canvas[y, x++] = char
+    //Iterate through codepoints in input string
+    for (index in 0 until cell.cell.content.unicodeLength) {
+      // TODO invisible chars, graphemes, etc.
+      val codePoint = cell.cell.content.codePointAt(index)
```

Comment:
I don't think this loop is correct. You are iterating over codepoint count but this function takes a character index.

It would need to be
```kotlin
var index = 0
while (index < cell.cell.content.length) {
  val codePoint = cell.cell.content.codePointAt(index)
  // ...
  index += Character.charCount(codePoint)
}
```

Reply:
Ooh, good spot (and a horrible API). I'll try to write a more complex test that demonstrates this and see if there's a nice alternative.

Reply:
You can use
```kotlin
cell.cell.content.codePoints().forEach { codePoint ->
  // ..
}
```
which is the Java `IntStream` API added to `CharSequence` in Java 8. You might need to cast it to `java.lang.String` first though...

Reply:
Ooh perfect, that's exactly what I was looking for :+1: